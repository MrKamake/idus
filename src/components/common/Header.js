import React from 'react';
import { Link } from 'react-router-dom';
import Logo from '../../images/logo.svg';
import './Header.scss';

const Header = () => {
  return (
    <div className="Header">
      <Link to="/">
        <img src={Logo} alt="Logo" />
      </Link>
      <ul>
        <Link to="/card">
          <li>Card</li>
        </Link>
        <Link to="/input">
          <li>Input</li>
        </Link>
      </ul>
    </div>
  );
};

export default Header;
