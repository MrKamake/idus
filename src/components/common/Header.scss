.Header {
  max-width: 60%;
  margin-left: auto;
  margin-right: auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
  position: relative;
  width: 100%;
  padding: 9px 0;
  img {
    display: flex;
    cursor: pointer;
    padding: 1px 2px;
  }
  a {
    text-decoration: none;
    color: black;
    text-align: center;
  }
  ul {
    display: flex;
    padding-left: 0;
    li {
      margin: 0 10px;
      list-style-type: none;
      font-size: 18px;
    }
  }
}
