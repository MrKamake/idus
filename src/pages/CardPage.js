import React from 'react';
import CardItem from '../components/card/CardItem';

const CardPage = () => {
  return (
    <div className="Card_Page">
      <CardItem additions="comment" />
      <CardItem additions="rating" />
      <CardItem />
    </div>
  );
};

export default CardPage;
