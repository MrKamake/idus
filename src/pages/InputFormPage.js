import React from 'react';
import InputFormContainer from '../containers/InputFormContainer';
import InputFormTemplate from '../components/input/InputFormTemplate';

const InputFormPage = () => {
  return (
    <InputFormTemplate>
      <InputFormContainer />
    </InputFormTemplate>
  );
};

export default InputFormPage;
