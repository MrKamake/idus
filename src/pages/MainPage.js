import React from 'react';
import { Link } from 'react-router-dom';
import './MainPage.scss'

const MainPage = () => {
  return (
    <div className="Main-Page">
      <Link to="/card">Card</Link>
      <Link to="/input">Input</Link>
    </div>
  );
};

export default MainPage;