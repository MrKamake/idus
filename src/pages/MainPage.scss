.Main-Page {
  padding-top: 15%;
  width: 90%;
  a {
    display: inline-block;
    margin: 5%;
    padding: 20% 30%;
    width: 40%;
    text-align: center;
    font-size: 1.5rem;
    text-decoration: none;
    color: white;
    background-color: gray;
    border-radius: 12px;
  }
}
