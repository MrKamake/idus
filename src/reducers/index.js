import { combineReducers } from 'redux';
import { GET_CARD_ITEMS, GET_REQUEST_ITEMS } from '../constants';

const initialState = {
  cardItems: [],
  requestItems: []
};

export const getDataReducer = (state = initialState, action) => {
  const { payload } = action;
  switch (action.type) {
    case GET_CARD_ITEMS:
      return { ...state, cardItems: payload };
    case GET_REQUEST_ITEMS:
      return { ...state, requestItems: payload };
    default:
      return state;
  }
};

export default combineReducers({ getDataReducer });
