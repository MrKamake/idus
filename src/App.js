import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import Header from './components/common/Header';
import MainPage from './pages/MainPage';
import CardPage from './pages/CardPage';
import InputFormPage from './pages/InputFormPage';

function App() {
  return (
    <div className="App">
      <Header />
      <Route exact path="/" render={() => <Redirect to="/main" />} />
      <Route path="/card" component={CardPage} />
      <Route path="/input" component={InputFormPage} />
      <Route exact path="/main" component={MainPage} />
    </div>
  );
}

export default App;
