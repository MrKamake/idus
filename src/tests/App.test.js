import React from 'react';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import { MemoryRouter } from 'react-router-dom';

import App from '../App';
import { getDataReducer } from '../reducers';
import Header from '../components/common/Header';
import MainPage from '../pages/MainPage';
import CardPage from '../pages/CardPage';
import InputFormPage from '../pages/InputFormPage';

describe('<App />', () => {
  describe('should be render according to the route', () => {
    let mockStore;
    let mockState;

    beforeEach(() => {
      mockStore = configureMockStore();
      mockState = { getDataReducer };
    });

    it('/ redirect to /main', () => {
      const wrapper = mount(
        <MemoryRouter initialEntries={['/']}>
          <App />
        </MemoryRouter>
      );
      expect(wrapper.find(Header)).toHaveLength(1);
      expect(wrapper.find(MainPage)).toHaveLength(1);
      expect(wrapper.find(CardPage)).toHaveLength(0);
      expect(wrapper.find(InputFormPage)).toHaveLength(0);
    });

    it('/card', () => {
      let store = mockStore(mockState);
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/card']}>
            <App />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(Header)).toHaveLength(1);
      expect(wrapper.find(MainPage)).toHaveLength(0);
      expect(wrapper.find(CardPage)).toHaveLength(1);
      expect(wrapper.find(InputFormPage)).toHaveLength(0);
    });

    it('/input', () => {
      let store = mockStore(mockState);
      const wrapper = mount(
        <Provider store={store}>
          <MemoryRouter initialEntries={['/input']}>
            <App />
          </MemoryRouter>
        </Provider>
      );
      expect(wrapper.find(Header)).toHaveLength(1);
      expect(wrapper.find(MainPage)).toHaveLength(0);
      expect(wrapper.find(CardPage)).toHaveLength(0);
      expect(wrapper.find(InputFormPage)).toHaveLength(1);
    });
  });
});
