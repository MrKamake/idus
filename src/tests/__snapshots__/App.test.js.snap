// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should be render according to the route matches snapshot 1`] = `
<Fragment>
  <Header />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    component={[Function]}
    path="/card"
  />
  <Route
    component={[Function]}
    path="/input"
  />
  <Route
    component={[Function]}
    exact={true}
    path="/main"
  />
</Fragment>
`;
